src/Libraries/Microsoft.Extensions.Http.Resilience/Resilience/HttpClientBuilderExtensions.Resilience.cs

```diff
@@ -4,82 +4,164 @@
-        if (provider is ByAuthorityPipelineKeyProvider)
+        if (provider is ByAuthorityStrategyKeyProvider)
         {
 #pragma warning disable S1075 // URIs should not be hardcoded - this URL is not used for any real request, nor in any telemetry
             using var request = new HttpRequestMessage(HttpMethod.Get, "https://localhost:123");
 #pragma warning restore S1075 // URIs should not be hardcoded
-            _ = provider.GetPipelineKey(request);
+            _ = provider.GetStrategyKey(request);
         }
     }
 
-    private static HttpResiliencePipelineBuilder AddHttpResiliencePipeline(this IHttpClientBuilder builder, string pipelineIdentifier)
+    private static IHttpResilienceStrategyBuilder AddHttpResilienceStrategy(
+        this IHttpClientBuilder builder,
+        string name,
+        Action<ResilienceStrategyBuilder<HttpResponseMessage>, ResilienceHandlerContext> configure)
+    {
+        var strategyName = StrategyNameHelper.GetName(builder.Name, name);
+        var key = new HttpKey(name, string.Empty);
+
+        _ = builder.Services.AddResilienceStrategy<HttpKey, HttpResponseMessage>(key, (builder, context) => configure(builder, new ResilienceHandlerContext(context)));
+
+        ConfigureHttpServices(builder.Services);
+
+        return new HttpResilienceStrategyBuilder(name, builder.Services);
+    }
+
+    private static void ConfigureHttpServices(IServiceCollection services)
     {
-        _ = builder.Services.ConfigureHttpFailureResultContext();
-        var pipelineName = PipelineNameHelper.GetPipelineName(builder.Name, pipelineIdentifier);
-        var pipelineBuilder = builder.Services.AddResiliencePipeline<HttpResponseMessage>(pipelineName);
+        // don't add any new service if this method is called multiple times
+        if (services.Contains(Marker.ServiceDescriptor))
+        {
+            return;
+        }
+
+        services.Add(Marker.ServiceDescriptor);
+
+        // This code configure the multi-instance support of the registry
+        _ = services.Configure<ResilienceStrategyRegistryOptions<HttpKey>>(options =>
+        {
+            options.BuilderNameFormatter = key => key.Name;
+            options.StrategyKeyFormatter = key => key.Key;
+            options.BuilderComparer = HttpKey.BuilderComparer;
+        });
 
-        return new HttpResiliencePipelineBuilder(pipelineBuilder);
+        _ = services
+            .AddExceptionSummarizer(b => b.AddHttpProvider())
+            .ConfigureFailureResultContext<HttpResponseMessage>((response) =>
+            {
+                if (response != null)
+                {
+                    var statusCodeName = ((int)response.StatusCode).ToInvariantString();
+                    if (string.IsNullOrEmpty(statusCodeName) || char.IsDigit(statusCodeName[0]))
+                    {
+                        statusCodeName = TelemetryConstants.Unknown;
+                    }
```

Comment:
What does this do? I don't understand. Convert enum to int, then to string and check if its not null or empty? Which I don't think it ever can be or if it's a digit? Can int as string ever not be a digit? I am very confused. Maybe I am missing something.

Reply:
It's indeed strange. I have just ported this code but it makes little sense.

Reply:
fixed ... hopefully